### PR TITLE
Add recording related information to NEST 3 transition guide.

### PR DESCRIPTION
**Pull request by jougs**
The new paragraphs contain all I want to say content-wise. My main concern is that I used all the markup (`*TEXT*`, `` `TEXT` ``) wrong. @jessica-mitchell: could you please have a brief look?

### CODE REVIEW

File: doc/guides/from_nest2_to_nest3.rst [hunk 1]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+the name of the recorded variable for each column as a descriptive
+meta-data header prior to writing any data.
+
+The `accumulator_mode` of the ``multimeter`` has been dropped, as it
+was not used by anyone to the best of our knowledge and supporting it
+made the code more complex and prone to errors. In case of high user
+demand, the functionality will be re-added in form of a recording
+backend.
+
+All details about the new infrastucture can be found in the guide on
```
Comment (hakonsbm) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
All details about the new infrastructure can be found in the guide on
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 2]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+*files*) were handled by a single C++ class. Due to the many different
+responsibilities and the resulting complexity of this class, extending
+and maintaining it was rather burdensome.
+
+With NEST 3.0 we replaced this single class by an extensible and
+modular infrastructure for handling recordings: each modality is now
+taken care of by a specific recording backend and each recorder can
+use one of them to handle its data.
+
+NEST 3.0 brings recording backends for all modalities supported
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
NEST 3.0 supports the same recording backends for all modalities 
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 3]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+already in NEST 2.x. If compiled with support for `SIONlib
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
as in NEST 2.x. If compiled with support for `SIONlib
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 4]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+useful on large clusters and supercomputers.
+
+Changes
+^^^^^^^
+
+In NEST 2.x, the recording modality was selected by either providing a
+list of modalities to the `record_to` property, or by setting one or
+more of the flags `to_file`, `to_memory`, or `to_screen` to *True*.
+
+In NEST 3.0, the individual flags are gone and the `record_to`
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
In NEST 3.0, the individual flags are gone, and the `record_to`
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 5]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+property now expects the name of the backend to use. Recording to
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
property now expects the name of the backend you want to use. Recording to
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 6]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+multiple modalities from a single device is not possible anymore and
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
multiple modalities from a single device is no longer possible. 
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 7]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+individual devices have to be created and configured if this
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
Individual devices have to be created and configured if this
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 8]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+functionality is needed.
+
+All following examples assume that the variable `mm` points to a
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
The following examples assume that the variable `mm` points to a
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 9]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+``multimeter`` instance, i.e. that ``mm = nest.Create('multimeter')``
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
``multimeter`` instance, i.e.,  ``mm = nest.Create('multimeter')``
```

File: doc/guides/from_nest2_to_nest3.rst [hunk 10]
```diff
@@ -1248,9 +1250,92 @@ probability and delay, and random weights from a normal distribution:
...
+  +------------------------------------------------------+--------------------------------+
+  | ::                                                   | ::                             |
+  |                                                      |                                |
+  |     nest.SetStatus(mm, {'to_file': True})            |     mm.record_to = "ascii"     |
+  |     nest.SetStatus(mm, {'to_screen': True})          |     mm.record_to = "screen"    |
+  |     nest.SetStatus(mm, {'to_memory': True})          |     mm.record_to = "memory"    |
+  |                                                      |                                |
+  +------------------------------------------------------+--------------------------------+
+
+The list of available backends can be retrieved using the following command:
```
Comment (jessica-mitchell) [inline doc/guides/from_nest2_to_nest3.rst]:
```suggestion
You can retrieve the list of available backends using the following command:
```
Nothing wrong with what you had, but I tend to prefer including some more active voice in the instructions

**Pull request merged**